File: interop-consumer-java/src/main/java/org/metacubed/samples/interop/java/service/ServiceImpl.java
package org.metacubed.samples.interop.java.service;

import org.metacubed.samples.interop.kotlin.model.InputData;
import org.metacubed.samples.interop.kotlin.model.OutputData;
import org.metacubed.samples.interop.kotlin.service.Service;
import org.metacubed.samples.interop.kotlin.util.Converter;

public class ServiceImpl implements Service {

    private final Converter converter = new Converter();
    private final InputData input;

    public ServiceImpl(InputData input) {

        this.input = input;
    }

    @Override
    public OutputData convert()  {

        return converter.convert(input);
    }
}

File: interop-consumer-kotlin/src/main/kotlin/org/metacubed/samples/interop/kotlin/service/ServiceImpl.kt
package org.metacubed.samples.interop.kotlin.service

import org.metacubed.samples.interop.kotlin.model.InputData
import org.metacubed.samples.interop.kotlin.util.Converter

internal class ServiceImpl(
    private val input: InputData,
    private val converter: Converter = Converter()
) : Service {

    override fun convert() = converter.convert(input)
}

